eng/deployment/product-construction-service-deploy.ps1

```diff
@@ -0,0 +1,81 @@
+# This script deploys the Product Construction Service, using the blue/green deployment pattern.
+# The script detirmines the color of the currently active revision, deactivates the old inactive revision, 
+# and deploys the new revision, switching all traffic to it if the health probes pass.
+param(
+    [Parameter(Mandatory=$true)][string]$resourceGroupName,
+    [Parameter(Mandatory=$true)][string]$containerappName,
+    [Parameter(Mandatory=$true)][string]$newImageTag,
+    [Parameter(Mandatory=$true)][string]$containerRegistryName,
+    [Parameter(Mandatory=$true)][string]$imageName,
+    [Parameter(Mandatory=$true)][string]$subscriptionName,
+    [Parameter(Mandatory=$true)][string]$containerappEnvironmentName
+)
+
+az extension add --name containerapp --upgrade
+
+Write-Host "Fetching all revisions to determine the active label"
+$containerappTraffic = az containerapp ingress traffic show --name $containerappName --resource-group $resourceGroupName | ConvertFrom-Json
+# find the currently active revision
+$activeRevision = $containerappTraffic | Where-Object { $_.weight -eq 100 } 
+
+Write-Host "Currently active revision: $($activeRevision.revisionName) with label $($activeRevision.label)"
+
+# detirmine the label of the inactive revision
+if ($activeRevision.label -eq "blue") {
+    $inactiveLabel = "green"
+} else {
+    $inactiveLabel = "blue"
+}
+
+Write-Host "Next revision will be deployed with label $inactiveLabel"
+Write-Host "Removing label $inactiveLabel from the inactive revision"
+# remove the label from the inactive revision
+$revisionRemovalOutput = az containerapp revision label remove --label $inactiveLabel --name $containerappName --resource-group $resourceGroupName 2>&1
+
+if ($revisionRemovalOutput -match "Please specify a label name with an associated traffic weight") {
+    Write-Host "Couldn't find a revision with label $inactiveLabel. Skipping deactivation of inactive revision"
+} 
+else
+{
+    Write-Host "Deactivating inactive revision"
+    # deactivate the inactive revision
+
+    $inactiveRevision = $containerappTraffic | Where-Object { $_.label -eq $inactiveLabel }
+
+    az containerapp revision deactivate --revision $inactiveRevision.revisionName --name $containerappName --resource-group $resourceGroupName
+}
+# deploy the new image
+$newImage = "$containerRegistryName.azurecr.io/$imageName`:$newImageTag"
+Write-Host "Deploying new image $newImage"
+az containerapp update --name $containerappName --resource-group $resourceGroupName --image $newImage --revision-suffix $newImageTag | Out-Null
```

Comment:
What happens if the image is not there? Will this be a visible message?

Reply:
Or why the `| Out-Null`?

Reply:
We`ll get an error then, the `Out-Null` only affects stdout. I added it because the output of these az cli commands is a full json view of the affected resource, which is quite long, and convolutes the output